cep-XXXX.md

```diff
@@ -0,0 +1,229 @@
+
+These constraints motivate the following set of rules for mapping conda packages to OCI repositories and tags. More explanation is given below in the Rational section and the example implementation illustrate how these rules work together to create a human-readable and easily parsable OCI repository `<name>` and `<tag>` for a conda package.
+
+#### Overall Form of the OCI Repository and Tag
+
+A conda package with its channel, subdir, and (optional) label MUST be mapped to an OCI `<repository>:<tag>` in one of two forms.
+
+The unhashed form of the OCI repository and tag for a conda package is:
+
+```
+<channel>/<subdir>/<OCI-encoded package name>:<OCI-encoded version, build and (optional) label in the form '<version>-<build>(-<label>)'>
+```
+
+In the hashed form, the OCI-form of the package name and the string composed of the OCI-forms of the version, build, and (optional) label (i.e., `<version>-<build>(-<label>)`) are each separately hashed via SHA1 and then replaced by the string `h<hexdigest of SHA1 hash>` as follows
+
+```
+<channel>/<subdir>/h<hexdigest of the SHA1 hash of OCI-encoded package name>:h<hexdigest of the SHA1 hash of the original OCI tag of the conda package>
+```
+
+The `<channel>` and `<subdir>` MUST be used as-is without modification. The OCI-forms of the package name, label, version, and build string are defined below.
+
+#### Encoding Package Names, Channels, and Subdirs into OCI Repository `<name>`s
+
+The package name MUST be encoded to OCI-form as follows:
+
+- If the package name starts with an `_`, the `_` MUST be replaced with `z`.
+- Otherwise, if the package name does NOT start with an `_`, the package name MUST be prepended by `c`.
+- If the combined string `<channel>/<subdir>/<OCI-encoded package name>` exceeds 128 characters in length, the OCI-encoded package name MUST be replaced by the SHA1 hash of the OCI-encoded package name in hexadecimal form as `h<SHA1 in hexidecimal>`.
+
+Package names in OCI-form which are not hashed can be decoded by reversing the encoding rules above.
+
+For example, the package name `_libgcc_mutex` is encoded to OCI-form as `zlibgcc_mutex`, and the package name `zlibgcc_mutex` is encoded to OCI-form as `czlibgcc_mutex`. A very long package name like `gblah000...000` would be encoded to its OCI-form `cgblah000...000`, then hashed via SHA1, yielding something like `h<hexdigest of SHA1 hash>`.
+
+The channel and subdir MUST be used as-is without modification.
+
+The channel, subdir and OCI-encoded package name (or hashed OCI-encoded package name if required) MUST be combined with `/` to form the OCI repository `<name>`
+
+```
+<channel>/<subdir>/<OCI-encoded package name>
+```
+
+#### Encoding Labels, Versions and Build Strings into OCI `<tag>`s
+
+The label, version, and build string are encoded to OCI-form as follows. First the following mapping rules MUST be applied to the label, version and build string in the order listed below:
+
+- `_` -> `__`
```

Comment:
Only single underscores are allowed as input, right? Because OCI only allows two consecutive underscores IIRC.

Reply:
Yep. Changed to `_U` per your suggestion!